server/data_common/cache.py

```diff
@@ -209,6 +209,11 @@ def get(self, cache_key: str, create_data_function: Optional[Callable] = None, c
             if cache_item:
                 cache_item.release()
 
+    def evict_by_key(self, cache_key: str):
+        evict = self.data.get(cache_key, None)
+        if evict:
+            self.evict_data(to_del=[(cache_key, evict)])
```

Comment:
Just curious, why does `evict_data` need anything more than the `cache_key`?

Reply:
it could probably be refactored to only take the cache_key (and use that to lookup the cache item) but currently it calls the attempt_delete function on the cache item in order to call the delte/cleanup on the datasets